plugins/editor/index.js

```diff
@@ -20,51 +20,24 @@ function editor(app, opts, done){
 
   var codeEditor;
   var outputConsole;
-  var buffer = new ConsoleBuffer();
-
-  var refreshQueued = null;
-  var lastRefresh = 0;
-  var refreshDelayMillis = 64;
-
-  var space = app.workspace;
-
-  function output(evt){
-    buffer.update(evt);
-
-    if(refreshQueued != null){
-      return;
-    }
-    if(lastRefresh < Date.now() - refreshDelayMillis){
-      refreshConsole();
-    }else{
-      refreshQueued = setTimeout(refreshConsole, refreshDelayMillis);
-    }
-  }
+  var state = {};
 
   function refreshConsole(){
+    const { buffer } = state;
     if(outputConsole){
       outputConsole.innerHTML = buffer.getConsoleHTML();
       outputConsole.scrollTop = outputConsole.scrollHeight;
     }
-    refreshQueued = null;
-    lastRefresh = Date.now();
   }
 
-  function clearOutput(){
-    buffer.clear();
-    if(outputConsole){
-      outputConsole.innerHTML = buffer.getConsoleHTML();
-    }
-    if(refreshQueued != null){
-      clearInterval(refreshQueued);
-      refreshQueued = null;
-    }
-    lastRefresh = 0;
+  function onChange() {
+    state = consoleStore.getState();
+    refreshConsole();
   }
 
-  output.clear = clearOutput;
+  consoleStore.listen(onChange);
 
-  app.expose('logger', output);
+  var space = app.workspace;
```

Comment:
space? used in a few places it seems: lines 52, 70, 72, 87, 95, 98.

i have removed some of these in my fluxEditor branch. i will be submitting the PR for that shortly.